maitake/src/wait/cell.rs

```diff
@@ -134,7 +134,9 @@ impl WaitCell {
                 test_trace!(state = ?actual, "was notified");
                 let waker = self.waker.with_mut(|waker| unsafe { (*waker).take() });
                 // Reset to the WAITING state by clearing everything *except*
-                // the closed bits (which must remain set).
+                // the closed bits (which must remain set). This `fetch_and`
+                // does *not* set the CLOSED bit if it is unset, it just doesn't
+                // clear it.
                 let state = test_dbg!(self.fetch_and(State::CLOSED, AcqRel));
                 // The only valid state transition while we were parking is to
                 // add the CLOSED bit.
```

Comment:
I think this was the actual comment that tripped me up, thinking that the only way that the `self.compare_exchange(State::PARKING, State::WAITING, AcqRel)` could fail would be a closed event.